applications/ShapeOptimizationApplication/python_scripts/custom_math.py

```
@@ -0,0 +1,422 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+#
+# ==============================================================================
+
+# Making KratosMultiphysics backward compatible with python 2.6 and 2.7
+from __future__ import print_function, absolute_import, division
+
+# Import additional libraries
+import math
+
+# ==============================================================================
+def SafeConvertVectorToMatrix(_L):
+    if IsVector(_L):
+        return [_L]
+    else:
+        return _L
```

should raise an error here